File: c#/5177.cs
using System;
using System.Text.RegularExpressions;

class Program
{
    static string sanitize(string str)
    {
        str = str.Trim().ToLower();
        str = new Regex(@"\s{2,}").Replace(str, " ");
        str = new Regex(@"[{\[]").Replace(str, "(");
        str = new Regex(@"[}\]]").Replace(str, ")");
        str = str.Replace(';', ',');
        str = new Regex(@"\s*([().,:])\s*").Replace(str, "$1");

        return str;
    }

    static void Main()
    {
        int K = int.Parse(Console.ReadLine());

        foreach (var i in Enumerable.Range(1, K))
        {
            Console.WriteLine($"Data Set {i}: {(sanitize(Console.ReadLine()) == sanitize(Console.ReadLine()) ? "equal" : "not equal")}");
            
            if (i < K)
            {
                Console.WriteLine("");
            }
        }
    }
}

File: python/5177.py
import re


def sanitize(string):
    string = string.strip().lower()
    string = re.sub(r"\s{2,}", " ", string)
    string = re.sub(r"[{\[]", "(", string)
    string = re.sub(r"[}\]]", ")", string)
    string = string.replace(";", ",")
    string = re.sub(r"\s*([().,:])\s*", r"\1", string)

    return string


for i in range(1, K := int(input()) + 1):
    print(f"Data Set {K}: {'equal' if sanitize(input()) == sanitize(input()) else 'not equal'}")

    if i < K:
        print("")
